.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      module:
+        description: "Maven (Sub)module to Release (e.g.,
+                  keycloak-extensions,
+                  authenticator-required-action,
+                  keycloak-extension-rest,
+                  user-management).
+                  Parent is the default"
+        required: false
+        default: "parent"
+      releaseVersion:
+        description: "Version to use for release."
+        required: true
+        default: "X.Y.Z"
+      developmentVersion:
+        description: "Version to use for next development cycle."
+        required: true
+        default: "X.Y.Z-SNAPSHOT"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Configure Git User
+        run: |
+          git config user.email "actions@github.com"
+          git config user.name "GitHub Actions"
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: 'maven'
+
+      - name: Install Parent POM (If necessary locally)
+        run: mvn install -N || echo "Parent-POM already exists or is getting downloaded from repo."
+
+
+      - name: Determine Build Scope
+        id: determine_scope
+        run: |
+          if [[ "${{ github.event.inputs.module }}" == "parent" ]]; then
+            echo "BUILD_MODULE=parent" >> $GITHUB_ENV
+            echo "BUILD_ARGS=" >> $GITHUB_ENV
+          else
+            echo "BUILD_MODULE=${{ github.event.inputs.module }}" >> $GITHUB_ENV
+            echo "BUILD_ARGS=-pl ${{ github.event.inputs.module }} -am" >> $GITHUB_ENV
+          fi
+
+      # ✅ Nur das gewählte Modul wird gebaut (Gesamtprojekt oder einzelnes Submodul)
+      - name: Build and Package Selected Module
+        run: |
+          if [[ "$BUILD_MODULE" == "parent" ]]; then
+            mvn clean package
+          else
+            mvn clean package $BUILD_ARGS
+          fi
+
+
+      - name: Release Selected Module
+        run: |
+          if [[ "$BUILD_MODULE" == "ROOT" ]]; then
+            echo "releasing parent with version ${{ github.event.inputs.releaseVersion }}"
+            mvn -B release:prepare release:perform \
+              -DpushChanges=false \
+              -DlocalCheckout=true \
+              -DreleaseVersion=${{ github.event.inputs.releaseVersion }} \
+              -DdevelopmentVersion=${{ github.event.inputs.developmentVersion }} \
+              -DautoVersionSubmodules=false \
```

Comment:
Are we just setting the version on the parent module in this case?
How do we ensure that version numbers of sub-modules are unique?

Let's say parent has version 1.0.0 and a sub-modul has version 2.3.4. If we make changes to the submodul and release the whole project via the parent, the parent may become version 1.1.0. But the submodule still has version 2.3.4. in this case. But it changed and therefore needs a new version number.